pydantic/_internal/_std_types_schema.py

```diff
@@ -67,58 +67,65 @@ def get_enum_core_schema(enum_type: type[Enum], config: ConfigDict) -> CoreSchem
-        except ValueError:
-            raise PydanticCustomError('enum', 'Input should be {expected}', {'expected': expected})
-
+    sub_type: typing_extensions.Literal['str', 'int', 'float'] | None = None
     if issubclass(enum_type, int):
-        # this handles `IntEnum`, and also `Foobar(int, Enum)`
-        js_updates['type'] = 'integer'
-        lax_schema = core_schema.no_info_after_validator_function(to_enum, core_schema.int_schema())
+        sub_type = 'int'
+        value_ser_type: core_schema.SerSchema = core_schema.simple_ser_schema('int')
     elif issubclass(enum_type, str):
         # this handles `StrEnum` (3.11 only), and also `Foobar(str, Enum)`
-        js_updates['type'] = 'string'
-        lax_schema = core_schema.no_info_after_validator_function(to_enum, core_schema.str_schema())
+        sub_type = 'str'
+        value_ser_type = core_schema.simple_ser_schema('str')
     elif issubclass(enum_type, float):
-        js_updates['type'] = 'numeric'
-        lax_schema = core_schema.no_info_after_validator_function(to_enum, core_schema.float_schema())
+        sub_type = 'float'
+        value_ser_type = core_schema.simple_ser_schema('float')
     else:
-        lax_schema = core_schema.no_info_plain_validator_function(to_enum)
+        # TODO this is an ugly hack, how do we trigger an Any schema for serialization?
+        value_ser_type = core_schema.plain_serializer_function_ser_schema(lambda x: x)
+
+    if cases:
+
+        def get_json_schema(schema: CoreSchema, handler: GetJsonSchemaHandler) -> JsonSchemaValue:
+            json_schema = handler(schema)
+            original_schema = handler.resolve_ref_schema(json_schema)
+            update_json_schema(original_schema, js_updates)
+            return json_schema
+
+        # we don't want to add the missing_func to the schema if it's the default one
+        default_missing_func = getattr(enum_type._missing_, '__func__', None) == Enum._missing_.__func__  # type: ignore
+        enum_schema = core_schema.enum_schema(
+            enum_type,
+            cases,
+            sub_type=sub_type,
+            missing_func=None if default_missing_func else enum_type._missing_,
+            ref=enum_ref,
+            metadata={'pydantic_js_functions': [get_json_schema]},
+        )
 
-    enum_schema = core_schema.lax_or_strict_schema(
-        lax_schema=lax_schema,
-        strict_schema=core_schema.json_or_python_schema(
-            json_schema=lax_schema, python_schema=core_schema.is_instance_schema(enum_type)
-        ),
-        ref=enum_ref,
-        metadata={'pydantic_js_functions': [get_json_schema]},
-    )
-    if config.get('use_enum_values', False):
-        enum_schema = core_schema.no_info_after_validator_function(attrgetter('value'), enum_schema)
+        if config.get('use_enum_values', False):
+            enum_schema = core_schema.no_info_after_validator_function(
+                attrgetter('value'), enum_schema, serialization=value_ser_type
```

Comment:
I presume adding the `serialization` schema here doesn't mess up the `use_enum_values` logic given that we apply that behavior during validation? Will just want to make sure tests don't fail...

Reply:
we should check, I don't think there are any tests for this case.

Reply:
I think this should test for that case: https://github.com/pydantic/pydantic/blob/e7baf5856d9d1e9aab1403b7da78149a8e3f3137/tests/test_types.py#L6420-L6436

Reply:
As in, that test is passing, so I think we're fine.